Update property metadata

andrewlee94 opened the pull request:
https://github.com/IDAES/idaes-pse/pull/995 changes how metadata is defined for property packages, and enforces standard naming conventions (or at least forces users to explicitly declare non-standard names). This revealed that some of our examples were using non-standard names, which this PR addresses.

This PR will likely fail without the changes in `idaes-pse`, but this will be necessary to test those changes.

## Proposed changes:
- Update HDA property package to use standard names.
- Updated custom compressor example properties to use more standard names (and drop some unused properties).

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

bpaul4 commented on pkg/idaes_examples/common/hda/hda_ideal_VLE.py:
```diff
@@ -376,44 +376,35 @@ def build(self):
-             'dh_vap': {'method': '_dh_vap', 'units': 'J/mol'},
-             'ds_vap': {'method': '_ds_vap', 'units': 'J/mol.K'}})
-
+                     'method': '_enenrgy_internal_mol_phase'},
```
Typo "enenrgy". I followed the commit history and this typo has been here since the original file was added, which then brings up whether there should have been a warning or error that indicated an incorrect method name.

andrewlee94 replied:
@bpaul it would be possible to verify the names, but we generally use the approach and "try it and let it fail then". In this case, it just means this property was never used by an example.

bpaul4 commented on src/Examples/Advanced/CustomUnitModels/methanol_param_VLE.py:
```diff
@@ -107,6 +107,7 @@ def build(self):
              3: ["H2", ("Vap", "Liq")],
              4: ["CH3OH", ("Vap", "Liq")]}
 
+        # Antoine coefficients assume pressure in mmHG and temperature in K
```
This comment line was duplicated somehow.

bpaul4 reviewed (commented):
Just a couple things I noticed, I will re-review after the idaes-pse changes are merged or if the tests pass without it.

andrewlee94 commented:
@bpaul4 I addressed the issues you found, and the test failures we are seeing as expected (due to the yet-to-be-merged changes in idaes-pse).

However, this PR needs to be merged before the changes in idaes-pse (despite the test failures). We have a chicken-and-egg problem where we have test failures on both sides due to these changes, so one side needs to be merged first with failures, and I think it is best to start here.

bpaul4 reviewed (approved):
Approving so the process can move forward.

bpaul4 commented:
FYI it's also possible to fix this PR's head to the corresponding idaes-pse PR branch, rather than merging to test it, if that would be more convenient.

andrewlee94 commented:
@bpaul4 That won't really work - the issue is that neither PR will pass tests until one of them has been merged, so we just have to merge one with failures. Even if we did change the head to test this PR, we would still have to undo the changes before we merge it in which case the failures would be back (and it takes over an hour to run the tests too).
